project_files/bank_stripped.js

```diff
@@ -0,0 +1,37 @@
+class Printer {     // Responsibility - Print account history
+    printStatement(account) {
+        console.log("Account Summary:\nDate || Credit || Debit || Balance");
+        account.transactions.forEach(transaction => {
+            console.log(transaction.join(" || "));
+        });
+    }
+}
+
+class Account {     // Responsibility - Store account details (balance and transaction history)
+    constructor() {
+        this._balance = 0;
+        this._transactions = [];
+    }
+
+    get balance() {return this._balance;}
+
+    set balance(amount) {this._balance = amount}
+
+    get transactions() {return this._transactions}
+
+    set transactions(amount) {this._transactions.unshift(amount)}
+}
+
+class ATM {         // Responsibility - Edit account balance and transaction history
```

Comment:
How many sources of truth are there for the balance amount? The account has a balance that's mutated, and each transactions keeps a copy of the balance.

Where possible, prefer 1 source of truth for values.
Where possible, prefer to calculate at runtime, rather mutate state.